kolibri/auth/test/test_users.py

```diff
@@ -26,33 +25,33 @@ class UserProxyManagerTestCase(TestCase):
     Checks that the BaseUser proxy models' default manager returns appropriate instances.
     """
     def setUp(self):
-        self.user = FacilityUser.objects.create(username="mike", first_name="Mike", last_name="Gallaspy")
-        self.user = FacilityUser.objects.create(username="brian", first_name="Brian", last_name="Gallaspy")
+        self.dataset = FacilityDataset.objects.create()
         self.fu_usernames = ["mike", "brian"]
-        self.do = DeviceOwner.objects.create(username="bar")
+        for username in self.fu_usernames:
+            FacilityUser.objects.create(username=username, dataset=self.dataset)
         self.do_usernames = ["bar"]
+        for username in self.do_usernames:
+            DeviceOwner.objects.create(username=username)
 
     def test_facility_user_manager_returns_facility_users(self):
-        base_users = BaseUser.objects.filter(username__in=self.fu_usernames)
-        facility_users = FacilityUser.objects.all()
-        self.assertEqual(len(base_users), len(facility_users))
+        facility_user = FacilityUser.objects.all()[0]
+        self.assertIsInstance(facility_user, FacilityUser)
 
     def test_fu_set_correct(self):
         """ continuation of test_facility_user_manager_returns_facility_users """
         base_users = BaseUser.objects.filter(username__in=self.fu_usernames)
         facility_users = FacilityUser.objects.all()
-        self.assertListEqual(sorted([u.id for u in base_users]), sorted([u.id for u in facility_users]))
+        self.assertEqual(set([u.id for u in base_users]), set([u.id for u in facility_users]))
 
     def test_device_owner_manager_returns_device_owners(self):
-        base_users = BaseUser.objects.filter(username__in=self.do_usernames)
-        device_owners = DeviceOwner.objects.all()
-        self.assertEqual(len(base_users), len(device_owners))
+        device_owner = DeviceOwner.objects.all()[0]
+        self.assertIsInstance(device_owner, DeviceOwner)
 
     def test_do_set_correct(self):
         """ continuation of test_device_owner_manager_returns_device_owners """
         base_users = BaseUser.objects.filter(username__in=self.do_usernames)
         device_owners = DeviceOwner.objects.all()
-        self.assertListEqual(sorted([u.id for u in base_users]), sorted([u.id for u in device_owners]))
+        self.assertEqual(set([u.id for u in base_users]), set([u.id for u in device_owners]))
```

Comment:
Ah, interesting -- so, protecting against duplicate records, then (otherwise, set comparison also does that). Sounds good, will revert.